glommio/src/channels/channel_mesh.rs

```diff
@@ -0,0 +1,312 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the
+// MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2020 Datadog, Inc.
+//
+use std::cell::Cell;
+use std::fmt::{self, Debug, Formatter};
+use std::sync::{Arc, RwLock};
+
+use futures::future;
+
+use crate::channels::local_channel::{self, *};
+use crate::channels::shared_channel::{self, *};
+use crate::{Local, Result};
+
+#[derive(Debug)]
+pub enum Sender<T: Send + Copy> {
+    Local(LocalSender<T>),
+    Connected(ConnectedSender<T>),
+}
+
+impl<T: Send + Copy> Sender<T> {
+    pub async fn send(&self, item: T) -> Result<(), T> {
+        match self {
+            Self::Local(sender) => sender.send(item).await,
```

Comment:
Okay, so are you using the `LocalChannel` just to have the option of sending this both locally or remotely?

What would the value be for using this over local channels?

Reply:
My intention was to provide a uniform interface in cases when users may want a local channel to implement staged event-driven architecture. But indeed those are minor cases and it introduces complexity with all the enums to unify the local and remote cases. I'm now considering removing the local support,  or at least make it optional. Which way would be better?

Reply:
For now I think we should keep it simple and remove the local case.
By using the same interface we're forcing `Send` on types that will pass through the local channel which is not good anyway.